R/plots.R

```diff
@@ -535,3 +544,190 @@ cumulative_value.plot <- function(data,
+  }
+
+  #Will plot the lift plot with the MDE and the power curve
+  if (!(market_ID %in% x$BestMarkets$ID)){
+    stop("Please enter a valid ID.")
+  }
+
+  Market <- x$BestMarkets %>% dplyr::filter(ID == market_ID)
+
+  locs_aux <- unlist(strsplit(stringr::str_replace_all(Market$location, ", ", ","), split = ","))
+  max_time <- max(x$parameters$data$time)
+
+  PowerPlot <- as.data.frame(x$PowerCurves %>% dplyr::filter(duration == Market$duration,
+                                                             location == Market$location))
+
+  CostPerLift <- as.numeric(PowerPlot %>%
+                              dplyr::filter(EffectSize != 0) %>%
+                              dplyr::mutate(AvgCost = Investment / abs(EffectSize)) %>%
+                              dplyr::summarise(mean(AvgCost)))
+
+  data_lifted <- x$parameters$data
+  data_lifted$Y[data_lifted$location %in% locs_aux &
+                  data_lifted$time >= max_time - Market$duration + 1] <-
+    data_lifted$Y[data_lifted$location %in% locs_aux &
+                    data_lifted$time >= max_time - Market$duration + 1] * (1 + Market$EffectSize)
+
+
+  lifted <- suppressMessages(GeoLift::GeoLift(
+    Y_id = "Y",
+    time_id = "time",
+    location_id = "location",
+    data = data_lifted,
+    locations = locs_aux,
+    treatment_start_time = max_time - Market$duration + 1,
+    treatment_end_time = max_time,
+    model = x$parameters$model,
+    fixed_effects = x$parameters$fixed_effects,
+    print = TRUE
+  ))
+
+  if (print_summary){
+    # message(paste0(
+    #   "##################################",
+    #   "\n#####   GeoLift Simulation   #####\n",
+    #   "##################################\n",
+    #   "\n** Simulating a ",
+    #   100*round(Market$EffectSize,2),
+    #   "% Effect Size **" ))
+    message(paste0(
+      "##################################",
+      "\n#####   GeoLift Simulation   #####\n",
+      "####  Simulating: ",
+      100*round(Market$EffectSize,2),
+      "% Lift  ####\n",
+      "##################################"))
+    print(summary(lifted))
+  }
+
+  PowerPlot_graph <- ggplot(PowerPlot, aes(x = EffectSize, y = power)) +
+    geom_smooth(formula = y ~ x, method = "loess", se = FALSE, aes(colour = "Smoothed Values")) +
+    geom_line(size = 0.62, alpha = 0.8, aes(colour = "Actual Values")) +
+    scale_x_continuous(sec.axis = sec_axis( ~ . * CostPerLift ,
+                                            breaks = scales::pretty_breaks(n = breaks_x_axis),
+                                            name = "Estimated Investment"),
+                       breaks = scales::pretty_breaks(n = breaks_x_axis)) +
+    ylim(0, 1) +
+    geom_hline(yintercept = 0.8, linetype = "dashed", color = "grey") +
+    scale_colour_manual(name="Power Curve", values=c("gray80", "#52854C")) +
+    labs( x = "Effect Size", y = "Power") +
+    theme_minimal() +
+    theme(plot.title = element_text(hjust = 0.5))
```

Comment:
Artur, maybe its worth calling the GeoLiftPower function here instead of redefining the plot object? What do you think?

Reply:
I left this one separate as I think it will be worth revisiting `GeoLiftPower()`'s status as active or superseded soon.